packages/react/toolbar/src/Toolbar.tsx

```diff
@@ -0,0 +1,246 @@
+
+const GROUP_NAME = 'ToolbarGroup';
+
+type ToolbarGroupOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarGroupPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarGroupOwnProps
+>;
+
+const ToolbarGroup = React.forwardRef((props, forwardedRef) => {
+  const { orientation } = useToolbarContext(GROUP_NAME);
+
+  return (
+    <Primitive
+      role="group"
+      selector={getSelector(GROUP_NAME)}
+      data-orientation={orientation}
+      {...props}
+      ref={forwardedRef}
+    />
+  );
+}) as ToolbarGroupPrimitive;
+
+ToolbarGroup.displayName = GROUP_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarLabel
+ * -----------------------------------------------------------------------------------------------*/
+
+const LABEL_NAME = 'ToolbarLabel';
+
+type ToolbarLabelOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarLabelPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarLabelOwnProps
+>;
+
+const ToolbarLabel = React.forwardRef((props, forwardedRef) => (
+  <Primitive selector={getSelector(LABEL_NAME)} {...props} ref={forwardedRef} />
+)) as ToolbarLabelPrimitive;
+
+ToolbarLabel.displayName = LABEL_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarSeparator
+ * -----------------------------------------------------------------------------------------------*/
+
+const SEPARATOR_NAME = 'ToolbarSeparator';
+
+type ToolbarSeparatorOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarSeparatorPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarSeparatorOwnProps
+>;
+
+const ToolbarSeparator = React.forwardRef((props, forwardedRef) => {
+  const { orientation } = useToolbarContext(GROUP_NAME);
+
+  return (
+    <Primitive
+      role="separator"
+      selector={getSelector(SEPARATOR_NAME)}
+      aria-orientation={orientation}
+      data-orientation={orientation}
+      {...props}
+      ref={forwardedRef}
+    />
+  );
+}) as ToolbarSeparatorPrimitive;
+
+ToolbarSeparator.displayName = SEPARATOR_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarItem
+ * -----------------------------------------------------------------------------------------------*/
+
+const ITEM_NAME = 'ToolbarItem';
+
+type ToolbarItemOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarItemPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarItemOwnProps
+>;
+
+const ToolbarItem = React.forwardRef((props, forwardedRef) => {
+  const { disabled, ...itemProps } = props as any;
+
+  const rovingFocusProps = useRovingFocus({
+    disabled: disabled,
```

Comment:
nit: looks like this could be the following:

```suggestion
    disabled,
```